tests/test_linter/test_ignore_types.py

```diff
@@ -0,0 +1,472 @@
+import yaml
+
+from linkml.linter.linter import Linter
+from linkml.utils.schema_builder import SchemaBuilder
+
+
+def test_class_empty_title_forbidden():
+    config = yaml.safe_load(
+        """
+rules:
+  no_empty_title:
+    level: error
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_class("MyClass")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    # print("\n")
+    # for m in messages:
+    #     print(m)
+
+    assert len(messages) == 1
+    assert any("Class 'MyClass' has no title" in m for m in messages)
+
+
+def test_class_empty_title_allowed():
+    config = yaml.safe_load(
+        """
+rules:
+  no_empty_title:
+    level: error
+    exclude_type:
+      - class_definition
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_class("MyClass")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+def test_slot_empty_title_forbidden():
+    config = yaml.safe_load(
+        """
+rules:
+  no_empty_title:
+    level: error
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_slot("my_slot")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 1
+    assert any("Slot 'my_slot' has no title" in m for m in messages)
+
+
+def test_slot_empty_title_allowed():
+    config = yaml.safe_load(
+        """
+rules:
+  no_empty_title:
+    level: error
+    exclude_type:
+      - slot_definition
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_slot("my_slot")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+def test_enum_empty_title_forbidden():
+    config = yaml.safe_load(
+        """
+rules:
+  no_empty_title:
+    level: error
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_enum("MyEnum")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 1
+    assert any("Enum 'MyEnum' has no title" in m for m in messages)
+
+
+def test_enum_empty_title_allowed():
+    config = yaml.safe_load(
+        """
+rules:
+  no_empty_title:
+    level: error
+    exclude_type:
+      - enum_definition
+"""
+    )
+
+    builder = SchemaBuilder()
+    builder.add_enum("MyEnum")
+
+    linter = Linter(config)
+    report = list(linter.lint(builder.schema))
+
+    messages = [p.message for p in report]
+
+    assert len(messages) == 0
+
+
+# todo PVs are not checked for titles yet
+
+
+def test_class_missing_recommendeds_forbidden():
```

Comment:
Recommend:

* Drop `test_*_missing_recommendeds_forbidden` functions (already tested in `test_rule_recommended.py`)
* Move `test_*_missing_recommendeds_allowed` functions (consider renaming) into `test_rule_recommended.py`